### DISCUSSION TITLE
FluentIcon syntax

### DISCUSSION THREAD

**Discussion by p1971** [microsoft/fluentui-blazor#1110] [General]
hi, just started looking at FluentUI for a new website - as a "lighter" alternative to MudBlazor (creates big and chunky ui's) ... looks nice and clean so far!

is there are reason a new instance of the icon is created, eg:
```
<FluentIcon Value="@(new Icons.Regular.Size24.Save())" />
```

rather than using a static accessor (or constant, like MudBlazor) for example ?

eg if I have a grid with action buttons for each row (same icon etc) then I'd imagine I'd be creating a lot of new icons etc whereas I could do something like this:
```
public static class MyIcons
{
    public static Icons.Regular.Size24.Save Save = new Icons.Regular.Size24.Save();
}

<FluentIcon Value="@MyIcons.Save" />
```

**Comment by dvoituron** [comment]
You can also use this syntax, but with the **Icon** attribute: `<FluentIcon Icon="@Icons.Regular.Size24.Shifts30Minutes" />` but in this case the dotnet compiler doesn't know that this icon is being used and the WASM trimming process will remove this icon from the final library.

In summary, 
- You can use `Icon` or `Value`attributes with a Server project.
- You should use `Value` attribute with a WASM project to avoid problem.

Like explain here: https://www.fluentui-blazor.net/Icon

**Reply by p1971** [reply]
Thanks - I was reading the https://www.fluentui-blazor.net/IconsAndEmoji page  and missed the Icon one you linked
